### PR TITLE
Remove specific plugin names in shared test case

### PR DESCRIPTION
**Pull request by panhaiqi**
### Details:
 - *Remove specific plugin names in shared test case*
 -  *Throw exception if device list in both ov::device::priorities() and device name (For example AUTO:GPU,CPU)*

### Tickets:
 - *CVS-130214*

### CODE REVIEW

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 1]
```diff
@@ -230,6 +229,8 @@ std::vector<ov::AnyMap> OVPropertiesTestsWithCompileModelProps::getROOptionalPro
         res.push_back({{ov::PropertyName(ov::device::type.name(), ov::device::type.mutability), nullptr}});
         res.push_back({{ov::PropertyName(ov::execution_devices.name(), ov::execution_devices.mutability), nullptr}});
         res.push_back({{ov::PropertyName(ov::available_devices.name(), ov::available_devices.mutability), nullptr}});
+        res.push_back(
```
Comment (sbalandi) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:232]: please move under the condition if (!is_sw_device) this property in the function OVPropertiesTestsWithCompileModelProps::getRWMandatoryPropertiesValues [str 271](https://github.com/openvinotoolkit/openvino/blob/2d22dd70c0f1fa783fa6c34a5d67b009396a7bc1/src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp#L271)

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 2]
```diff
@@ -91,7 +91,6 @@ void OVPropertiesTestsWithCompileModelProps::SetUp() {
     }
 
     if (target_device == std::string(ov::test::utils::DEVICE_MULTI) ||
```
Comment (sbalandi) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:93]: this whole if can be removed now, adding MULTI_DEVICE_PRIORITIES is now handled by the --config

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 3]
```diff
@@ -82,7 +82,7 @@ void OVPropertiesTestsWithCompileModelProps::SetUp() {
 
     std::string::size_type pos = temp_device.find(":", 0);
     std::string hw_device;
-    if (pos == std::string::npos) {
+    if (pos == std::string::npos && temp_device != "AUTO") {
```
Comment (sbalandi) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:85]: Why ? Which device will be used in tests as target_device ?

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp (hunk 2 above)
Comment (peterchen-intel) [reply src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:93]: @sbalandi Will you update the conformance test for sw plugin to use --config_path= in the conformance test in pre-commit/commit and nightly? Or use AUTO:TEMPLATE for test?

Comment (sbalandi) [reply src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:93]: --config_path have already been used in nightly validation; conformance with sw plugins is not run in pre_commit, only cpu conformance

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 4]
```diff
@@ -81,8 +81,10 @@ void OVPropertiesTestsWithCompileModelProps::SetUp() {
     std::tie(temp_device, properties) = this->GetParam();
 
     std::string::size_type pos = temp_device.find(":", 0);
-    std::string hw_device;
-    if (pos == std::string::npos) {
+    std::string hw_device = "";
+    if (temp_device == "AUTO") {
+        target_device = temp_device;
+    } else if (pos == std::string::npos) {
```
Comment (peterchen-intel) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:87]: 1. If Find : in temp_device
2. Find target_device and set device::priorites
3. Else
4. target_device = temp_device

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 5]
```diff
@@ -81,19 +81,14 @@ void OVPropertiesTestsWithCompileModelProps::SetUp() {
...
-    std::string hw_device;
-    if (pos == std::string::npos) {
-        target_device = temp_device;
-        hw_device = temp_device;
+    std::string hw_device = "";
+    if (pos != std::string::npos) {
+        if (!target_device.empty()) {
+            target_device = temp_device.substr(0, pos);
+            compileModelProperties = {ov::device::priorities(target_device)};
+        }
```
Comment (peterchen-intel) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:89]:
```suggestion
        target_device = temp_device.substr(0, pos);
        compileModelProperties = {ov::device::priorities(target_device)};
```

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 6]
```diff
@@ -81,19 +81,14 @@ void OVPropertiesTestsWithCompileModelProps::SetUp() {
...
-        target_device = temp_device.substr(0, pos);
-        hw_device = temp_device.substr(++pos, std::string::npos);
-    }
-
-    if (target_device == std::string(ov::test::utils::DEVICE_MULTI) ||
-        target_device == std::string(ov::test::utils::DEVICE_AUTO) ||
-        target_device == std::string(ov::test::utils::DEVICE_HETERO) ||
-        target_device == std::string(ov::test::utils::DEVICE_BATCH)) {
+        target_device = temp_device;
         compileModelProperties = {ov::device::priorities(hw_device)};
```
Comment (peterchen-intel) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:92]:
```suggestion
```

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 7]
```diff
@@ -81,19 +81,14 @@ void OVPropertiesTestsWithCompileModelProps::SetUp() {
     std::tie(temp_device, properties) = this->GetParam();
 
     std::string::size_type pos = temp_device.find(":", 0);
-    std::string hw_device;
-    if (pos == std::string::npos) {
-        target_device = temp_device;
-        hw_device = temp_device;
+    std::string hw_device = "";
```
Comment (peterchen-intel) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:84]:
```suggestion
```

**Review by sbalandi — APPROVED** [review]

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 8]
```diff
@@ -81,20 +81,11 @@ void OVPropertiesTestsWithCompileModelProps::SetUp() {
...
-
-    if (target_device == std::string(ov::test::utils::DEVICE_MULTI) ||
-        target_device == std::string(ov::test::utils::DEVICE_AUTO) ||
-        target_device == std::string(ov::test::utils::DEVICE_HETERO) ||
-        target_device == std::string(ov::test::utils::DEVICE_BATCH)) {
-        compileModelProperties = {ov::device::priorities(hw_device)};
+        compileModelProperties = {ov::device::priorities(target_device)};
+    } else {
+        target_device = temp_device;
     }
```
Comment (peterchen-intel) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:90]:
```suggestion
```

**Comment by peterchen-intel** [comment]
@sbalandi Please check again, I think there is no reason to extract device list here now.

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 9]
```diff
@@ -77,25 +77,6 @@ std::string OVPropertiesTestsWithCompileModelProps::getTestCaseName(testing::Tes
 
 void OVPropertiesTestsWithCompileModelProps::SetUp() {
     SKIP_IF_CURRENT_TEST_IS_DISABLED();
```
Comment (peterchen-intel) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:79]:
```suggestion
    SKIP_IF_CURRENT_TEST_IS_DISABLED();
    std::tie(target_device, properties) = this->GetParam();
```

**Comment by peterchen-intel** [comment]
> > @sbalandi Please check again, I think there is no reason to extract device list here now.
> 
> yes, I agree with you, but I just realized that this construction is still needed for `ov func tests` for `sw plugins`, so maybe we need to keep it, but use something like that:
> 
> ```
>     std::string::size_type pos = temp_device.find(":", 0);
>     if (pos != std::string::npos) {
>         target_device = temp_device.substr(0, pos);
>         if (target_device == std::string(ov::test::utils::DEVICE_MULTI) ||
>             target_device == std::string(ov::test::utils::DEVICE_AUTO) ||
>             target_device == std::string(ov::test::utils::DEVICE_HETERO) ||
>             target_device == std::string(ov::test::utils::DEVICE_BATCH)) {
>             compileModelProperties = {ov::device::priorities(temp_device.substr(++pos, std::string::npos))};
>         }
>     }
> ```

Need to insert ov::device::priorities() to compileModelProperties instead of replacing it with only one element ov::device::priorities(). Throw exception if there is already ov::device::priorities() in compileModelProperties.

**Comment by peterchen-intel** [comment]
std::string::size_type pos = temp_device.find(":", 0);
    if (pos != std::string::npos) {
        target_device = temp_device.substr(0, pos);
        if (ov::device::priorities.name() is not in compileModelProperties) {
             insert ov::device::priorities(temp_device.substr(++pos, std::string::npos)) to compileModelProperties
        } else {
throw exception/fail the test case with information.
        }
    }

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 10]
```diff
@@ -81,20 +81,16 @@ void OVPropertiesTestsWithCompileModelProps::SetUp() {
...
-        target_device == std::string(ov::test::utils::DEVICE_AUTO) ||
-        target_device == std::string(ov::test::utils::DEVICE_HETERO) ||
-        target_device == std::string(ov::test::utils::DEVICE_BATCH)) {
-        compileModelProperties = {ov::device::priorities(hw_device)};
+        std::cout << compileModelProperties.size() << std::endl;
+        for (auto& it : compileModelProperties) {
+            OPENVINO_ASSERT(it.first == ov::device::priorities.name(),
+                            "there is already ov::device::priorities() in compileModelProperties");
+        }
+        compileModelProperties.insert(ov::device::priorities(temp_device.substr(++pos, std::string::npos)));
```
Comment (peterchen-intel) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:91]: If "properties" will be merged to compileModelProperties, need to check "properties" as well.

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 11]
```diff
@@ -431,6 +427,7 @@ OVPropertiesTestsWithCompileModelProps::getWrongRWOptionalPropertiesValues(
 
 TEST_P(OVCheckSetIncorrectRWMetricsPropsTests, ChangeIncorrectProperties) {
     std::vector<ov::PropertyName> supported_properties;
+    std::cout << "-------------------- " << target_device << std::endl;
```
Comment (zhaixuejun1993) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:430]: I suppose we do not need this print

File: src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp [hunk 12]
```diff
@@ -81,20 +81,16 @@ void OVPropertiesTestsWithCompileModelProps::SetUp() {
...
         target_device = temp_device.substr(0, pos);
-        hw_device = temp_device.substr(++pos, std::string::npos);
-    }
-
-    if (target_device == std::string(ov::test::utils::DEVICE_MULTI) ||
-        target_device == std::string(ov::test::utils::DEVICE_AUTO) ||
-        target_device == std::string(ov::test::utils::DEVICE_HETERO) ||
-        target_device == std::string(ov::test::utils::DEVICE_BATCH)) {
-        compileModelProperties = {ov::device::priorities(hw_device)};
+        std::cout << compileModelProperties.size() << std::endl;
```
Comment (zhaixuejun1993) [inline src/tests/functional/plugin/shared/src/behavior/ov_plugin/properties_tests.cpp:86]: remove

**Review by yangwang201911 — APPROVED** [review]

**Review by peterchen-intel — APPROVED** [review]

**Review by WeldonWangwang — APPROVED** [review]

**Review by iefode — APPROVED** [review]

**Pull request merged**